tests/components/history_stats/test_config_flow.py

```diff
@@ -44,6 +44,9 @@ async def test_form(
     result = await hass.config_entries.flow.async_configure(
         result["flow_id"],
         {
+            CONF_ENTITY_ID: "binary_sensor.test_monitored",
```

Comment:
Yeah I was kind of disappointed I had to do this to pass the testing, was initially hoping it would be less invasive.

If I mark the disabled fields as vol.Required, I have to pass them into this function. 

If I mark the disabled fields as vol.Optional, (and don't pass them in here), they don't get passed through the config+options flow and the assert fails at the end. 

Will have to think about this some more.

Reply:
Ok trying a bit different approach.

- Make readonly fields vol.Optional instead of vol.Required.
- Change frontend to delete the readonly fields before sending data back to backend.
- Change backend to not delete optional/readonly fields when they are not provided by the step flow (which they never will be provided).

With this can revert all the changes to the test.